free_fleet_adapter/free_fleet_adapter/nav2_robot_adapter.py

```diff
@@ -0,0 +1,290 @@
+        # TODO(ac): parameterize the service call timeout
+        replies = self.zenoh_session.get(
+            namespacify('navigate_to_pose/_action/get_result', self.name),
+            zenoh.Queue(),
+            value=req.serialize(),
+            # timeout=0.5
+        )
+        for reply in replies.receiver:
+            try:
+                # Deserialize the response
+                rep = NavigateToPose_GetResult_Response.deserialize(
+                    reply.ok.payload
+                )
+                self.node.get_logger().debug(f'Result: {rep.status}')
+                if rep.status == GoalStatus.STATUS_EXECUTING:
+                    return False
+                elif rep.status == GoalStatus.STATUS_SUCCEEDED:
+                    self.node.get_logger().info(
+                        f'Navigation goal {self.nav_goal_id} reached'
+                    )
+                    return True
+                else:
+                    self.node.get_logger().error(
+                        f'Navigation goal {self.nav_goal_id} status '
+                        f'{rep.status}')
+                    return True
+            except Exception as e:
+                self.node.get_logger().debug(
+                    'Received (ERROR: "{}"): {}: {}'
+                    .format(
+                        reply.err.payload.decode('utf-8'),
+                        type(e),
+                        e
+                    ))
+                continue
+
+    def update(self, state):
+        activity_identifier = None
+        if self.execution:
+            # TODO(ac): use an enum to record what type of execution it is,
+            # whether navigation or custom executions
+            if self.nav_goal_id is not None and self._is_navigation_done():
+                self.execution.finished()
+                self.execution = None
+                self.nav_goal_id = None
+            else:
+                activity_identifier = self.execution.identifier
+
+        self.update_handle.update(state, activity_identifier)
+
+    def make_callbacks(self):
+        return rmf_easy.RobotCallbacks(
+            lambda destination, execution: self.navigate(
+                destination, execution
+            ),
+            lambda activity: self.stop(activity),
+            lambda category, description, execution: self.execute_action(
+                category, description, execution
+            )
+        )
+
+    def navigate(self, destination, execution):
+        self.execution = execution
+        self.node.get_logger().info(
+            f'Commanding [{self.name}] to navigate to {destination.position} '
+            f'on map [{destination.map}]'
+        )
+
+        if destination.map != self.map:
+            self.node.get_logger().error(
+                f'Destination is on map [{destination.map}], while robot '
+                f'[{self.name}] is on map [{self.map}]'
+            )
+            return
```

Comment:
While this is unlikely, we could trigger a replan instead of returning with nothing to ensure that RMF sends in the next reasonable command. Otherwise the execution object remains waiting to be marked `finished()` without acknowledging that this command is invalid.